ui/apps/platform/src/Containers/Collections/RuleSelector/ByLabelSelector.tsx

```diff
@@ -0,0 +1,207 @@
+        if (selector.rules.every(({ key, values }) => key && values.every((value) => value))) {
+            selector.rules.push({ operator: 'OR', key: '', values: [''] });
+            handleChange(entityType, selector);
+        }
+    }
+
+    function onAddLabelValue(ruleIndex: number) {
+        const selector = cloneDeep(scopedResourceSelector);
+        const rule = selector.rules[ruleIndex];
+
+        // Only add a new form row if there are no blank entries
+        if (rule.values.every((value) => value)) {
+            rule.values.push('');
+            handleChange(entityType, selector);
+        }
+    }
+
+    function onDeleteValue(ruleIndex: number, valueIndex: number) {
+        if (!scopedResourceSelector.rules[ruleIndex]) {
+            return;
+        }
+
+        const newSelector = cloneDeep(scopedResourceSelector);
+
+        if (newSelector.rules[ruleIndex].values.length > 1) {
+            newSelector.rules[ruleIndex].values.splice(valueIndex, 1);
+            handleChange(entityType, newSelector);
+        } else if (newSelector.rules.length > 1) {
+            // This was the last value, so drop the rule
+            newSelector.rules.splice(ruleIndex, 1);
+            handleChange(entityType, newSelector);
+        } else {
+            // This was the last value in the last rule, so drop the selector
+            handleChange(entityType, {});
+        }
+    }
+
+    return (
+        <>
+            {scopedResourceSelector.rules.map((rule, ruleIndex) => {
+                const keyValidationError = validationErrors?.rules?.[ruleIndex];
+                const keyValidation =
+                    typeof keyValidationError === 'string' || keyValidationError?.key
+                        ? ValidatedOptions.error
+                        : ValidatedOptions.default;
+                return (
+                    <div key={rule.key}>
+                        {ruleIndex > 0 && (
+                            <Flex
+                                className="pf-u-pt-md pf-u-pb-xl"
+                                spaceItems={{ default: 'spaceItemsNone' }}
+                                alignItems={{ default: 'alignItemsCenter' }}
+                            >
+                                <Label variant="outline" isCompact>
+                                    and
+                                </Label>
+                                <span
+                                    style={{
+                                        borderBottom:
+                                            '2px dashed var(--pf-global--Color--light-300)',
+                                        flex: '1 1 0',
+                                    }}
+                                />
+                            </Flex>
+                        )}
+
+                        <Flex>
+                            <Flex className="pf-u-flex-grow-1">
+                                <FormGroup
```

Comment:
Similar to another comment, we might need `fieldId` prop for `FormGroup` and `id` prop for `AutoCompleteSelect` element for both accessibility and integration tests. Also, what about additional values after the first.